central/cve/node/datastore/datastore_impl.go

```diff
@@ -0,0 +1,201 @@
+package datastore
+
+import (
+	"context"
+
+	"github.com/gogo/protobuf/types"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/central/cve/common"
+	"github.com/stackrox/rox/central/cve/node/datastore/internal/index"
+	"github.com/stackrox/rox/central/cve/node/datastore/internal/search"
+	"github.com/stackrox/rox/central/cve/node/datastore/internal/store"
+	"github.com/stackrox/rox/central/role/resources"
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/sac"
+	pkgSearch "github.com/stackrox/rox/pkg/search"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	vulnRequesterOrApproverSAC = sac.ForResources(
+		sac.ForResource(resources.VulnerabilityManagementRequests),
+		sac.ForResource(resources.VulnerabilityManagementApprovals),
+	)
+
+	accessAllCtx = sac.WithAllAccess(context.Background())
+)
+
+type datastoreImpl struct {
+	storage  store.Store
+	indexer  index.Indexer
+	searcher search.Searcher
+
+	cveSuppressionLock  sync.RWMutex
+	cveSuppressionCache common.CVESuppressionCache
+}
+
+func (ds *datastoreImpl) buildSuppressedCache() error {
+	query := pkgSearch.NewQueryBuilder().AddBools(pkgSearch.CVESuppressed, true).ProtoQuery()
+	suppressedCVEs, err := ds.searcher.SearchRawCVEs(accessAllCtx, query)
+	if err != nil {
+		return errors.Wrap(err, "searching suppress CVEs")
+	}
+
+	ds.cveSuppressionLock.Lock()
+	defer ds.cveSuppressionLock.Unlock()
+	for _, cve := range suppressedCVEs {
+		ds.cveSuppressionCache[cve.GetId()] = common.SuppressionCacheEntry{
+			SuppressActivation: cve.GetSnoozeStart(),
+			SuppressExpiry:     cve.GetSnoozeExpiry(),
+		}
+	}
+	return nil
+}
+
+func (ds *datastoreImpl) Search(ctx context.Context, q *v1.Query) ([]pkgSearch.Result, error) {
```

Comment:
wdyt if we just embed the search into the datastore like:

type datastoreImpl struct {
  search.Searcher
}

then we don't even need these functions which are just passed through

Reply:
I prefer not to do any specialization for dackbox-related stores. It makes auto-generation tedious. If this is the pattern that can be followed everywhere, then it can very well come as an independent central-wide task.